umh-core/cmd/main.go

```diff
@@ -65,15 +70,24 @@ func main() {
 
 	// Start the control loop
 	controlLoop := control.NewControlLoop()
-	go SystemSnapshotLogger(ctx, controlLoop)
+	systemSnapshot := new(fsm.SystemSnapshot)
+	communicationState := communication_state.CommunicationState{
+		Watchdog:        watchdog.NewWatchdog(ctx, time.NewTicker(time.Second*10), true),
+		InboundChannel:  make(chan *models.UMHMessage, 100),
+		OutboundChannel: make(chan *models.UMHMessage, 100),
+		ReleaseChannel:  config.Agent.ReleaseChannel,
+	}
+	go SystemSnapshotLogger(ctx, controlLoop, systemSnapshot)
```

Comment:
You need to ensure that the ptr passed here and to enableBackendConnection is guarded